frontend/src/app/components/containers/complaints/webeoc-complaint-updates/webeoc-complaint-update-list.tsx

```diff
@@ -0,0 +1,128 @@
+import { FC, useEffect, useRef, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../../../hooks/hooks";
+import { getWebEOCUpdates, selectWebEOCComplaintUpdates } from "../../../../store/reducers/complaints";
+import { WebEOCComplaintUpdateDTO } from "../../../../types/app/complaints/webeoc-complaint-update";
+import { formatDate, formatTime } from "../../../../common/methods";
+
+type Props = {
+  complaintIdentifier: string;
+};
+
+export const WebEOCComplaintUpdateList: FC<Props> = ({ complaintIdentifier }) => {
+  const dispatch = useAppDispatch();
+  const complaintUpdates = useAppSelector(selectWebEOCComplaintUpdates);
+  const [expandedUpdates, setExpandedUpdates] = useState<Record<string, boolean>>({});
+  const [showLinks, setShowLinks] = useState<Record<string, boolean>>({});
+  const descriptionRefs = useRef<Record<string, HTMLDivElement | null>>({});
+
+  useEffect(() => {
+    dispatch(getWebEOCUpdates(complaintIdentifier));
+  }, [complaintIdentifier, dispatch]);
+
+  const toggleExpand = (id: string) => {
+    setExpandedUpdates((prev) => ({
+      ...prev,
+      [id]: true,
+    }));
+  };
+
+  useEffect(() => {
+    Object.keys(descriptionRefs.current).forEach((id) => {
+      const element = descriptionRefs.current[id];
+      if (element && element.scrollHeight > element.clientHeight) {
+        setShowLinks((prev) => ({
+          ...prev,
+          [id]: true,
+        }));
+      }
+    });
+  }, [complaintUpdates]);
+
+  return (
+    <>
+      {complaintUpdates && complaintUpdates.length > 0 ? (
```

Comment:
I would avoid a ternary statement here and instead suggest using a short circuit instead so as to avoid adding the "" to the markup

 ```
 {complaintUpdates && complaintUpdates.length !== 0 && (<>component goodness</>)}
 ```
If the ternary is needed, in your else block instead of an empty string use a fragment. This will prevent the empty string from being rendered
```
 {complaintUpdates && complaintUpdates.length !== 0 ? (<>component goodness</>) : <></> }
```

Reply:
Good call, I'll make it so